src/main/java/cachiluis/model/Subject.java

```diff
@@ -0,0 +1,44 @@
+package cachiluis.model;
+
+import datastructures.arraylist.MyArrayList;
+
+public class Subject {
+
+    private Teacher teacher;
+    private String subjectName;
+    private MyArrayList<Kardex> subjectKardexes;
+
+    public Subject(String subjectName, Teacher teacher) {
+        this.subjectName = subjectName;
+        this.teacher = teacher;
+        this.subjectKardexes = new MyArrayList<>();
+    }
+
+    public Teacher getTeacher() {
+        return teacher;
+    }
+
+    public void setTeacher(Teacher teacher) {
+        this.teacher = teacher;
+    }
+
+    public Kardex getKardex(String kardexId) {
+        for (int i = 0; i < subjectKardexes.size(); i++) {
+            Kardex index = subjectKardexes.get(i);
+            if (index.toString() == kardexId) {
```

Comment:
I know you need to identify your entities but that shouldn't be done with toString method. You can add an abstract parent super class and define an attribute id or something. I leave the rest to your criteria